### PR TITLE
[22.05] Improved migration scripts, documentation

### PR DESCRIPTION
**Pull request by jdavcs**
This is #14537 and #14723 backported to 22.05. Please see those PRs for description.

Also addresses #14680.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/model/migrations/README.md [hunk 1]
```diff
@@ -1,5 +1,513 @@
...
+
+#### Creating new revisions
+
+To create a revision for the galaxy model:
+
+`./run_alembic.sh revision --head=gxy@head -message "your description"`
+
+To create a revision for the install model:
+
+`./run_alembic.sh revision --head=tsi@head -message "your description"`
```
Comment (davelopez) [inline lib/galaxy/model/migrations/README.md]:
```suggestion
`./run_alembic.sh revision --head=gxy@head --message "your description"`

To create a revision for the install model:

`./run_alembic.sh revision --head=tsi@head --message "your description"`
```

Comment (jdavcs) [reply lib/galaxy/model/migrations/README.md]: Thank you  - good catch!

**Comment by jdavcs** [comment]
This is ready. Two additional commits fix a typo in a commit from this PR. Packages test failure unrelated.

**Comment by mvdbeek** [comment]
Thanks @jdavcs!

**Comment by nsoranzo** [comment]
@jdavcs Sorry for checking this late, I'm trying to merge forward but noticed this PR seems to undo some of the work made in https://github.com/galaxyproject/galaxy/pull/14715 , in particular commit https://github.com/galaxyproject/galaxy/pull/14715/commits/ab40d01ceb53471411d9a3d21aa092605abf40ce , which reduced the content duplication in `lib/galaxy/model/migrations/README.md` .
Is this correct?

**Comment by jdavcs** [comment]
@nsoranzo Yes, it does. #14730 will redo it. (when we decided that the right thing to do was backporting all these changes, that other commit had been merged already).

If you like, I can forward-merge it myself (I was expecting it might be not a straightforward merge).

**Comment by nsoranzo** [comment]
It doesn't seem to me that #14730 will redo the changes I mentioned, since it doesn't touch `lib/galaxy/model/migrations/README.md` at all?

**Comment by jdavcs** [comment]
> It doesn't seem to me that #14730 will redo the changes I mentioned, since it doesn't touch `lib/galaxy/model/migrations/README.md` at all?

It does: https://github.com/galaxyproject/galaxy/pull/14730/commits/13cd8531f19d429d58c0da1f2e838eb4dc575949

**Comment by nsoranzo** [comment]
OK, #14730 needed a rebase, looks fine now!

**Pull request merged**